core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -911,6 +911,23 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
   def currentPersistenceIds(entityType: String, afterId: Option[String], limit: Long): Source[String, NotUsed] =
     queryDao.persistenceIds(entityType, afterId, limit)
 
+  /**
+   * Load the last event for the given `persistenceId` up to the given `toSeqNr`.
+   *
+   * @param persistenceId
+   *   The persistence id to load the last event for.
+   * @param toSeqNr
+   *   The sequence number to load the last event up to.
+   * @param includeDeleted
+   *   Whether to include deleted events.
+   */
+  def loadLastEvent(
+      persistenceId: String,
+      toSeqNr: Long,
+      includeDeleted: Boolean): Future[Option[ClassicEventEnvelope]] = {
```

Comment:
Let's use the typed.EventEnvelope instead, see currentEventsByPersistenceIdTyped and use deserializeBySliceRow.

`includeDeleted` is a rather internal thing that shouldn't be needed in public api. We can always use false for this.